spot-ml/src/test/scala/org/apache/spot/proxy/ProxyWordCreationTest.scala

```diff
@@ -0,0 +1,287 @@
+package org.apache.spot.proxy
+
+
+import org.apache.log4j.{Level, LogManager}
+import org.apache.spot.SuspiciousConnectsArgumentParser.SuspiciousConnectsConfig
+import org.apache.spot.proxy.ProxySchema.Word
+import org.apache.spot.testutils.TestingSparkContextFlatSpec
+import org.scalatest.Matchers
+
+
+class ProxyWordCreationTest extends TestingSparkContextFlatSpec with Matchers {
+
+  val testConfigProxy = SuspiciousConnectsConfig(analysis = "proxy",
+    inputPath = "",
+    feedbackFile = "",
+    duplicationFactor = 1,
+    topicCount = 20,
+    hdfsScoredConnect = "",
+    threshold = 1.0d,
+    maxResults = 1000,
+    outputDelimiter = "\t",
+    ldaPRGSeed = None,
+    ldaMaxiterations = 20,
+    ldaAlpha = 1.02,
+    ldaBeta = 1.001)
+
+  "proxy word creation" should "return the correct word given the rules to form the word" in {
+
+    val logger = LogManager.getLogger("ProxyWordCreation")
+    logger.setLevel(Level.WARN)
+
+    // case class ProxyInput has the form:
+    // 1  p_date:String,
+    // 2  p_time:String,      <- currently used for feature creation
+    // 3  clientip:String,
+    // 4  host:String,        <- currently used for feature creation
+    // 5  reqmethod:String,   <- currently used for feature creation
+    // 6  useragent:String,   <- currently used for feature creation
+    // 7  resconttype:String, <- currently used for feature creation
+    // 8  duration:Int,
+    // 9  username:String,
+    // 10 webcat:String,
+    // 11 referer:String,
+    // 12 respcode:String,    <- currently used for feature creation
+    // 13 uriport:Int,
+    // 14 uripath:String,
+    // 15 uriquery:String,
+    // 16 serverip:String,
+    // 17 scbytes:Int,
+    // 18 csbytes:Int,
+    // 19 fulluri:String)     <- currently used for feature creation
+
+    val record1 = ProxyInput(
```

Comment:
two suggestions.... since there are so many records identified by number (record1, record2, record3 etc)
maybe the names could be more informative like "put202", "put230" , "get230" or even "put202withHighEntropyURL" (or whatever exactly is being tested for) so a person reading the test can tell why  "record3" is supposed to behave  in a certain way

the other suggestion is that since you gave a field schema above, there's no good reason not to flatten the record definitions to fewer lines